core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -351,7 +354,30 @@ public void invalidateTable(SessionContext context, TableIdentifier ident) {}
   @Override
   public Table registerTable(
       SessionContext context, TableIdentifier ident, String metadataFileLocation) {
-    throw new UnsupportedOperationException("Register table is not supported");
+    checkIdentifierIsValid(ident);
+
+    Preconditions.checkArgument(
+        metadataFileLocation != null && !metadataFileLocation.isEmpty(),
+        "Cannot register an empty metadata file location as a table");
```

Comment:
If the metadata location is null, this should not state that it was empty. Instead, use the more standard form for error messages: `"Invalid metadata file location: %s"`. Printing the location shows that it is null or empty.